src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SchemaUtils.java

```diff
@@ -0,0 +1,538 @@
+  public static final String BQ_CHANGELOG_FIELD_NAME_NUMBER_OF_RECORDS_IN_TRANSACTION =
+    "_metadata_spanner_number_of_records_in_transaction";
+  public static final String BQ_CHANGELOG_FIELD_NAME_NUMBER_OF_PARTITIONS_IN_TRANSACTION =
+    "_metadata_spanner_number_of_partitions_in_transaction";
+  public static final String BQ_CHANGELOG_FIELD_NAME_BIGQUERY_COMMIT_TIMESTAMP =
+    "_metadata_big_query_commit_timestamp";
+
+  private static final String INFORMATION_SCHEMA_TABLE_NAME = "TABLE_NAME";
+  private static final String INFORMATION_SCHEMA_COLUMN_NAME = "COLUMN_NAME";
+  private static final String INFORMATION_SCHEMA_SPANNER_TYPE = "SPANNER_TYPE";
+  private static final String INFORMATION_SCHEMA_ORDINAL_POSITION = "ORDINAL_POSITION";
+  private static final String INFORMATION_SCHEMA_CONSTRAINT_NAME = "CONSTRAINT_NAME";
+  private static final String INFORMATION_SCHEMA_ALL = "ALL";
+
+  /**
+   * @return the fields that are only used intermediately in the pipeline and are not corresponding
+   * to Spanner columns.
+   */
+  public static Set<String> getBigQueryIntermediateMetadataFieldNames() {
+    Set<String> fieldNames = new HashSet<>();
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ERROR);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_RETRY_COUNT);
+    fieldNames.add(BQ_CHANGELOG_FIELD_NAME_ORIGINAL_PAYLOAD_JSON);
+    return fieldNames;
+  }
+
+  /**
+   * @return a map where the key is the table name and the value is the SpannerTable object of the
+   * table name.
+   */
+  public static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, String changeStreamName) {
+    final Set<String> spannerTableNames = getSpannerTableNamesTrackedByChangeStreams(
+      databaseClient, changeStreamName);
+
+    final Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName =
+      getSpannerColumnNamesExplicitlyTrackedByChangeStreamsByTableName(
+        databaseClient, changeStreamName);
+
+    return getSpannerTableByName(databaseClient, spannerTableNames,
+      spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName);
+  }
+
+  private static Map<String, SpannerTable> getSpannerTableByName(
+    DatabaseClient databaseClient, Set<String> spannerTableNames,
+    Map<String, Set<String>> spannerColumnNamesExplicitlyTrackedByChangeStreamByTableName) {
+
+    StringBuilder queryInformationSchemaColumnsSqlStringBuilder = new StringBuilder(
+      "SELECT TABLE_NAME, COLUMN_NAME, ORDINAL_POSITION, SPANNER_TYPE "
+        + "FROM INFORMATION_SCHEMA.COLUMNS");
+
+    // Skip the columns of the tables that are not tracked by Change Streams.
+    if (!spannerTableNames.isEmpty()) {
+      queryInformationSchemaColumnsSqlStringBuilder.append(" WHERE");
+
+      for (String tableName : spannerTableNames) {
+        queryInformationSchemaColumnsSqlStringBuilder.append(" TABLE_NAME=\"");
+        queryInformationSchemaColumnsSqlStringBuilder.append(tableName);
```

Comment:
These should be within a parameterized query

Reply:
Did you mean "WHERE TABLE IN ('t1', 't2')"? It doesn't seem we can use array after "IN".

Reply:
`WHERE TABLE IN UNNEST (@array)`

Reply:
Fixed